initramfs/src/image.rs

```diff
@@ -0,0 +1,168 @@
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{Cursor, Read},
+    os::unix::fs::PermissionsExt,
+};
+
+use anyhow::{anyhow, Result};
+use bytes::Bytes;
+use cpio::{newc::Builder, write_cpio};
+use libflate::gzip::{Decoder, Encoder};
+use log::{debug, info};
+use serde::Deserialize;
+use tar::Archive;
+
+#[derive(Deserialize, Debug, Clone)]
+#[allow(non_snake_case)]
+pub struct LayerMetadata {
+    pub blobSum: String,
+}
+
+#[derive(Clone)]
+pub struct Layer {
+    pub metadata: LayerMetadata,
+    pub content: Cursor<Bytes>,
+}
+
+pub struct Image {
+    name: String,
+    tag: String,
+    pub layers: Vec<Layer>,
+}
+
+impl Image {
+    pub fn new(full_name: &str) -> Result<Self> {
+        // Extract the registry and tag from the image
+        let mut parts = full_name.split(':');
+        let name = parts.next().ok_or_else(|| anyhow!("Invalid image name"))?;
+        let tag = parts.next().unwrap_or("latest");
+
+        // If the registry doesn't contain a '/', it's in "library/", meaning it's from the docker hub official images
+        let image = Self {
+            name: if !name.contains('/') {
+                format!("library/{}", name)
+            } else {
+                name.to_string()
+            },
+            tag: tag.to_string(),
+            layers: Vec::new(),
+        };
+
+        debug!("Successfully parsed image : {:?}", image.get_name());
+
+        Ok(image)
+    }
+
+    pub fn get_name(&self) -> &str {
```

Comment:
getter in Rust are usually not prefixed with `get`. Please rename to `name`

Reply:
fixed!